evdns.c

```diff
@@ -1723,6 +1938,12 @@ struct evdns_server_port *
 evdns_add_server_port_with_base(struct event_base *base, evutil_socket_t socket, int flags, evdns_request_callback_fn_type cb, void *user_data)
 {
 	struct evdns_server_port *port;
+	int sock_type;
+	ev_socklen_t sock_type_len = sizeof(sock_type);
+	/* Check that socket is a UDP or TCP socket */
+	if (getsockopt(socket, SOL_SOCKET, SO_TYPE, (void*)&sock_type, &sock_type_len)
+			|| (sock_type != SOCK_DGRAM && sock_type != SOCK_STREAM))
+		return NULL;
```

Comment:
Hm, obtaining socket type from fd implicitly is not clean, separate interfaces should be provided

Reply:
Done.